iOS/MobileStarter/react-app/src/IModelPicker.tsx

```diff
@@ -0,0 +1,139 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import { MobileCore } from "@itwin/mobile-sdk-core";
+import { IconImage, useIsMountedRef } from "@itwin/mobile-ui-react";
+import { Project } from "@itwin/projects-client";
+import { IModelApp, NativeApp } from "@itwin/core-frontend";
+import { IModelsClient, MinimalIModel } from "@itwin/imodels-client-management";
+import { AccessTokenAdapter } from "@itwin/imodels-access-frontend";
+import { LocalBriefcaseProps } from "@itwin/core-common";
+import { fileSizeString, presentError, ButtonProps, HubScreenButton, HubScreenButtonListProps, HubScreenButtonList } from "./Exports";
+
+export interface IModelInfo {
+  minimalIModel: MinimalIModel;
+  briefcase?: LocalBriefcaseProps;
+}
+
+interface IModelButtonProps extends Omit<ButtonProps, "title"> {
+  modelInfo: IModelInfo;
+  onCacheDeleted?: (modelInfo: IModelInfo) => void;
+};
+
+function IModelButton(props: IModelButtonProps) {
+  const { modelInfo, onClick, onCacheDeleted } = props;
+  const { minimalIModel, briefcase } = modelInfo;
+  const isMountedRef = useIsMountedRef();
+
+  const getTitle = (title: string, briefcase: LocalBriefcaseProps | undefined) => {
+    if (!briefcase) return title;
+    return title + " (" + fileSizeString(briefcase.fileSize) + ")";
+  }
+
+  return <HubScreenButton
+    title={getTitle(minimalIModel.displayName, briefcase)}
+    onClick={onClick}
+  >
+    {briefcase && <div className="delete-button" onClick={async (e) => {
```

Comment:
Maybe pull this out into a separate function instead of declaring it inline?

Reply:
Done